### DISCUSSION TITLE
请教大佬，TabControl中切换不同的TabItem后，为啥绑定的数据会为null？

### DISCUSSION THREAD

**Discussion by rainteen** [BYJRK/DotNet-Discussions#380] [WPF 开发]
我在一个 View中包含了一个TabControl控件，这个TabControl控件的两个 TabItem页签里都绑定了此View对应的ViewModel中同一个依赖属性。并且在ViewModel初始化后已经准备好了依赖数据，debug不为null，View显示默认的第一个TabItem时数据可以正常展示，但切换到另一个TabItem后 绑定数据为 null了，很是不解。
求大佬指点迷津。

**Comment by BYJRK** [comment]
贴一下源代码吧。另外你说的切换到另一个后绑定数据为 `null`，指的是另一个 `TabItem` 绑定的属性为 `null`，还是说 viewmodel 中对应的属性的值也被修改成了 `null`？在切换回前一个 `TabItem` 后现象怎么样？

**Reply by rainteen** [reply]
非常感谢大佬回复。

**Comment by rainteen** [comment]
Sorry啊，最后确定是我在不同的 TabItem里放的自定义控件数据绑定问题，一开始一直以为是 TabItem切换导致的数据为null。
ViewModel里的属性值一直不为null的。代码比较复杂，设计到自定义控件和一些业务逻辑，就不贴了。
